Validation/SiPixelPhase1HitsV/src/SiPixelPhase1HitsV.cc

```diff
@@ -0,0 +1,266 @@
+    histo[TOF_R].fill(globalR, tof, id, &iEvent);
+    histo[TOF_Z].fill(globalZ, tof, id, &iEvent);
+    histo[ELOSS].fill(energyLoss, id, &iEvent);
+    histo[ENTRY_EXIT_X].fill(entryExitX, id, &iEvent);
+    histo[ENTRY_EXIT_Y].fill(entryExitY, id, &iEvent);
+    histo[ENTRY_EXIT_Z].fill(entryExitZ, id, &iEvent);
+    histo[LOCAL_X].fill(localX, id, &iEvent);
+    histo[LOCAL_Y].fill(localY, id, &iEvent);
+    histo[LOCAL_Z].fill(localZ,  id, &iEvent);
+    histo[LOCAL_PHI].fill(localPhi,  id, &iEvent);
+    histo[LOCAL_ETA].fill(localEta,  id, &iEvent);
+  }
+
+  // get high forward info
+  for (it = forwardHighInput->begin(); it != forwardHighInput->end(); ++it) {
+    auto id = DetId(it->detUnitId());
+    const GeomDetUnit * det=(const GeomDetUnit*)tracker->idToDetUnit( id );
+    GlobalPoint gpos=det->toGlobal(it->localPosition());
+
+    float tof = it->timeOfFlight();
+    float globalEta = gpos.eta();
+    float globalPhi = gpos.phi();
+    float globalR = gpos.mag();
+    float globalZ = gpos.z();
+
+    float energyLoss = it->energyLoss();
+
+    float entryExitX = ( it->entryPoint().x() - it->exitPoint().x() );
+    float entryExitY = ( it->entryPoint().y() - it->exitPoint().y() );
+    float entryExitZ = std::abs( it->entryPoint().z() - it->exitPoint().z() );
+
+    float localX = it->localPosition().x();
+    float localY = it->localPosition().y();
+    float localZ = it->localPosition().z();
+    float localPhi = it->localPosition().phi();
+    float localEta = it->localPosition().eta();
+
+    histo[TOF_ETA].fill(globalEta, tof, id, &iEvent);
+    histo[TOF_PHI].fill(globalPhi, tof, id, &iEvent);
+    histo[TOF_R].fill(globalR, tof, id, &iEvent);
+    histo[TOF_Z].fill(globalZ, tof, id, &iEvent);
+    histo[ELOSS].fill(energyLoss, id, &iEvent);
+    histo[ENTRY_EXIT_X].fill(entryExitX, id, &iEvent);
+    histo[ENTRY_EXIT_Y].fill(entryExitY, id, &iEvent);
+    histo[ENTRY_EXIT_Z].fill(entryExitZ, id, &iEvent);
+    histo[LOCAL_X].fill(localX, id, &iEvent);
+    histo[LOCAL_Y].fill(localY, id, &iEvent);
+    histo[LOCAL_Z].fill(localZ,  id, &iEvent);
+    histo[LOCAL_PHI].fill(localPhi,  id, &iEvent);
+    histo[LOCAL_ETA].fill(localEta,  id, &iEvent);
+  }
+
+
+  // Sim Hit efficiency info
+  edm::Handle< edm::View<reco::Track> > trackCollectionH;
+  iEvent.getByToken(tracksToken_, trackCollectionH);
+  const edm::View<reco::Track>&  tC = *(trackCollectionH.product()); 
+
+  edm::Handle<TrackingParticleCollection> TPCollectionH;
+  iEvent.getByToken(tpToken_,TPCollectionH);
+
+  edm::Handle<reco::TrackToTrackingParticleAssociator> theHitsAssociator;
+  iEvent.getByToken(trackAssociatorByHitsToken_,theHitsAssociator);
+  if ( theHitsAssociator.isValid() ) {
+    associatorByHits = theHitsAssociator.product();
+  }
+
+  edm::Handle<edm::SimTrackContainer> simTrackCollection;
+  iEvent.getByToken(simTracksToken_, simTrackCollection);
+  if ( simTrackCollection.isValid() ) {
+    simTC = simTrackCollection.product();
```

Comment:
This survives and leaks outside analyze(). Please remove. The variable looks to be unused.